### PR TITLE
Show popup if renaming a device fails

### PR DESCRIPTION
**Pull request by andresmanz**
- I moved pulseaudio_rename_success_cb into menu_info to keep the UI stuff out of pulseaudio_action
- I added ui_errordialog() to keep message dialog creation separated (don't know if it should be stored in the glade file, too)

I'm pretty sure you won't like this for several reasons. But hey, it's github and I'm here to learn!

### CODE REVIEW

File: src/menu_info.c [hunk 1]
```diff
@@ -549,15 +549,31 @@ void menu_info_item_rename_dialog(menu_info_item_t* mii)
         char* name = g_strstrip(g_strdup(entry));
 
         if(!g_str_equal(name, mii->desc))
-            pulseaudio_rename(mii, name);
+            pulseaudio_rename(mii, name, menu_info_pulseaudio_rename_success_cb);
```
Comment (christophgysin) [inline src/menu_info.c]: There is no need to pass this in.

File: src/menu_info.c [hunk 2]
```diff
@@ -549,15 +549,31 @@ void menu_info_item_rename_dialog(menu_info_item_t* mii)
...
 
         g_free(name);
     }
 
     gtk_widget_hide(GTK_WIDGET(dialog));
-
```
Comment (christophgysin) [inline src/menu_info.c]: Try to avoid including unnecessary changes to keep the diff as small as possible. Do cleanups like this in separate commits.

File: src/menu_info.c [hunk 3]
```diff
@@ -549,15 +549,31 @@ void menu_info_item_rename_dialog(menu_info_item_t* mii)
...
 }
 
+void menu_info_pulseaudio_rename_success_cb(pa_context* c, int success, void* userdata)
```
Comment (christophgysin) [inline src/menu_info.c]: Keep the existing callback, and make this a simple function that displays an error dialog. Then you are not forced to inherit the pulseaudio specific callback signature.

File: src/menu_info.c [hunk 4]
```diff
@@ -549,15 +549,31 @@ void menu_info_item_rename_dialog(menu_info_item_t* mii)
...
+    // TODO: try to autoload module-device-manager?
+    if(!success)
+    {
+        GtkMessageDialog* dialog = ui_errordialog();
+        gtk_message_dialog_format_secondary_text(dialog,
+                "Failed to rename %s '%s'!\nIs module-device-manager loaded?",
+                menu_info_type_name(mii->menu_info->type), mii->name);
+
+        gtk_dialog_run(dialog);
+        gtk_widget_destroy(dialog);
```
Comment (christophgysin) [inline src/menu_info.c]: Both of the above lines generate warnings. Always use the casting macros to convert to the right type.

See also: http://openbooks.sourceforge.net/books/wga/gtk.html#AEN194

File: src/menu_info.c [hunk 5]
```diff
@@ -549,15 +549,31 @@ void menu_info_item_rename_dialog(menu_info_item_t* mii)
...
+    }       
+}
```
Comment (christophgysin) [inline src/menu_info.c]: The above lines contain traling whitespace. Configure your editor to highlight those, and remove them.

File: src/pulseaudio_action.c [hunk 6]
```diff
@@ -99,23 +100,13 @@ void pulseaudio_rename(menu_info_item_t* mii, const char* name)
...
     pa_operation_unref(o);
 }
 
-void pulseaudio_rename_success_cb(pa_context *c, int success, void *userdata)
-{
-    menu_info_item_t* mii = userdata;
-
-    // TODO: try to autoload module-device-manager?
-    if(!success)
-        g_warning("failed to rename %s '%s'! module-device-manager loaded?\n",
```
Comment (christophgysin) [inline src/pulseaudio_action.c]: I would keep this callback, but have it call a function that the displays the dialog.

File: src/ui.c [hunk 7]
```diff
@@ -98,3 +98,11 @@ GtkEntry* ui_renamedialog_entry()
 {
     return (GtkEntry*) gtk_builder_get_object(builder, "entry");
 }
+
+GtkMessageDialog* ui_errordialog()
+{
+    GtkMessageDialog* dialog = gtk_message_dialog_new(NULL, GTK_DIALOG_MODAL,
+            GTK_MESSAGE_INFO, GTK_BUTTONS_OK, "An error occured");
+    gtk_window_set_title(GTK_WINDOW(dialog), "pasystray");
```
Comment (christophgysin) [inline src/ui.c]: This should be done in glade.

**Comment by christophgysin** [comment]
I created a commit with the suggested changes:
https://github.com/christophgysin/pasystray/commit/c538cae66dabda4d636697c5a376307445569529

The error dialog still needs to be moved to glade. Give it a try!

If not, I'll try to finish it in the next few days.

**Comment by andresmanz** [comment]
Thank you very much for taking the time to correct and explain - that was very enlightening. I'm going to adopt your suggestions in the future as best I can.

I'll do so and create a new PR for the error_dialog branch.
